### PR TITLE
add more versions for libsdl*

### PR DESCRIPTION
**Pull request by SFGrenade**
mostly github versions

* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/l/libsdl_image/xmake.lua [hunk 1]
```diff
@@ -38,7 +52,11 @@ package("libsdl_image")
     on_install(function (package)
         local configs = {"-DSDL2IMAGE_SAMPLES=OFF", "-DSDL2IMAGE_TESTS=OFF"}
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
-        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        if package:version():le("2.6.2") then
+            table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        else
+            table.insert(configs, "-DPNG_SHARED=" .. (package:config("shared") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/l/libsdl_image/xmake.lua:58]: why? png? no this option, only `SDL2IMAGE_PNG_SHARED`

we should use BUILD_SHARED_LIBS

https://github.com/libsdl-org/SDL_image/blob/36cbe2624d233e5cbc5d88427e0f6b0d4219caa2/CMakeLists.txt#L68

Comment (SFGrenade) [reply packages/l/libsdl_image/xmake.lua:58]: oh, then i just got confused by https://github.com/libsdl-org/SDL_image/compare/release-2.6.2...release-2.6.3#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL488-R488

**Comment by SFGrenade** [comment]
for some reason the windows CIs fail to clone sdl_image?

**Comment by SFGrenade** [comment]
gonna see if i add all the cmake options as xmake options

**Comment by waruqi** [comment]
The sdl_image git will clone all the submodules, which is very slow and unstable. We should use tar urls whenever possible.

**Comment by SFGrenade** [comment]
ah ok

**Comment by SFGrenade** [comment]
apparently pkgconf gets "MT" from somewhere but it wants it as lowercase (windows exclusive)

**Pull request merged**